### PR TITLE
Standardize on 'hyperparameter' usage

### PR DESCRIPTION
**Pull request by Crissman**
## Motivation
Found a few instances of 'hyper parameter' or 'hyper-parameter' and want to standardize on 'hyperparameter'.

## Description of the changes
Find deviants. Forcibly correct.

### CODE REVIEW

File: optuna/samplers/_tpe/sampler.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ class TPESampler(BaseSampler):
 
     For further information about TPE algorithm, please refer to the following papers:
 
-    - `Algorithms for Hyper-Parameter Optimization
+    - `Algorithms for Hyperparameter Optimization
```
Comment (toshihikoyanase) [inline optuna/samplers/_tpe/sampler.py:79]: This `Hyper-Parameter` is in the title of the [paper](https://papers.nips.cc/paper/2011/file/86e8f7ab32cfd12577bc2619bc635690-Paper.pdf), and we cannot change it. Could you revert this change?
```suggestion
    - `Algorithms for Hyper-Parameter Optimization
```

**Review by toshihikoyanase — APPROVED** [review]
LGTM!

**Comment by toshihikoyanase** [comment]
@keisuke-umezawa Could you review this PR, please?

**Review by keisuke-umezawa — APPROVED** [review]
